bitbybit-tests/tests/bitenum_tests.rs

```diff
@@ -243,3 +243,121 @@ fn cfg_in_enum_values() {
     #[cfg(feature = "test123")]
     assert_eq!(Foo::new_with_raw_value(u2::new(3)), Err(3));
 }
+
+#[test]
+fn enum_with_range_u8() {
+    #[derive(Debug, PartialEq)]
+    #[bitenum(u8, exhaustive: false)]
+    enum Foo {
+        Variant1 = 0x05,
+
+        #[range(0x10..=0x3D)]
+        RangeVariantHex(u8),
+
+        Variant2 = 73,
+
+        #[range(100..=150)]
+        RangeVariantDec(u8),
```

Comment:
We just discussed this, but I'll repeat in case someone else is curious as well. There is a problem with this definition.

Let's say you have this bitenum:

```
enum Foo {
  Option1 = 0,
  #[range(1..255)]
  Option2(u8)
}
```

Nothing will stop you from making Foo::Option2(0). Are we ok with that? Can we stop someone from making these in any way?